app/validators/declaration_date_validator.rb

```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+class DeclarationDateValidator < ActiveModel::EachValidator
+  RCF3339_DATE_REGEX = /\A\d{4}-\d{2}-\d{2}T(\d{2}):(\d{2}):(\d{2})([\.,]\d+)?(Z|[+-](\d{2})(:?\d{2})?)?\z/i.freeze
```

Comment:
can't we just use:
`validates_format_of: :declaration_date, with: /\A\d{4}-\d{2}-\d{2}T(\d{2}):(\d{2}):(\d{2})([\.,]\d+)?(Z|[+-](\d{2})(:?\d{2})?)?\z/i`

Reply:
it was there originally, I just moved it to a separate one as we may reuse it for other classes and it looks quite long in the ```Base```